src/hostapi/asio/pa_asio.cpp

```diff
@@ -1871,24 +1874,50 @@ static PaError ValidateAsioSpecificStreamInfo(
 {
     if( streamInfo )
     {
-        if( streamInfo->size != sizeof( PaAsioStreamInfo )
-                || streamInfo->version != 1 )
+        switch( streamInfo->version )
         {
+        case 0:
+            /* NOTE: Struct version 0 is invalid, possibly indicative of uninitialized data. */
             return paIncompatibleHostApiSpecificStreamInfo;
+
+        case 1:
+            /* NOTE: V1 structure's size is smaller by one pointer. */
+            if( streamInfo->size < sizeof( PaAsioStreamInfo ) - sizeof(PaAsio_MessageCallback*) )
+                return paIncompatibleHostApiSpecificStreamInfo;
+            break;
+
+        case 2:
+            if( streamInfo->size != sizeof( PaAsioStreamInfo ) )
+                return paIncompatibleHostApiSpecificStreamInfo;
+            break;
+
+        default:
+            /* Newer versions of the struct must be larger. */
```

Comment:
```suggestion
            /* Newer versions of the struct must not be smaller. */
```

It's ok if the sizes are equal, and note that this can happen even if you add a field to a struct, depending on the alignof of the struct.